src/sentry/integrations/jira_server/integration.py

```diff
@@ -235,24 +241,48 @@ def get_pipeline_views(self):
     def build_integration(self, state):
         install = state['installation_data']
         access_token = state['access_token']
-        hostname = urlparse(install['url']).netloc
+
+        external_id = '%s:%s' % (urlparse(install['url']).netloc, install['consumer_key'])
+        webhook_secret = sha1_text(install['private_key']).hexdigest()
+
+        credentials = {
+            'consumer_key': install['consumer_key'],
+            'private_key': install['private_key'],
+            'access_token': access_token['oauth_token'],
+            'access_token_secret': access_token['oauth_token_secret'],
+        }
+        # Create the webhook before the integration record exists
+        # so that if it fails we don't persist a broken integration.
```

Comment:
If creating the integration fails what happens to the webhook? Why would the integration creation fail? And what circumstances would a user be able to create the integration but not the webhook? Is it a permissions issue?

Reply:
The webhook could fail because of networking issues. Jira-Server will likely be behind firewalls and I wanted to make sure we could reach the target jira install before saving the integration record to our database.